sklearn/kernel_approximation.py

```diff
@@ -258,16 +223,16 @@ def fit(self, X, y=None):
         """Set parameters."""
         X = check_array(X, accept_sparse='csr')
         if self.sample_interval is None:
-            # See reference, figure 2 c)
-            if self.sample_steps == 1:
-                self.sample_interval_ = 0.8
-            elif self.sample_steps == 2:
-                self.sample_interval_ = 0.5
-            elif self.sample_steps == 3:
-                self.sample_interval_ = 0.4
+            if not isinstance(self.preset_sample_intervals, dict):
+                raise ValueError("Preset sample intervals not defined,"
+                                 " base class method call?")
```

Comment:
It means that one of the possible (but not the only) causes for the error ("Preset sample intervals not defined") is that base class is used without proper subclassing (defining preset_sample_intervals).